WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationViewModel.swift

```diff
@@ -63,11 +78,176 @@ final class CommentModerationViewModel: ObservableObject {
     }
 
     func didTapLike() {
-        switch state {
-        case .approved(let liked):
-            state = .approved(liked: !liked)
-        default:
-            break
+        let initialIsLiked = comment.isLiked
+
+        guard let siteID = siteID else {
+            self.state = .approved(liked: initialIsLiked)
+            return
+        }
+
+        if initialIsLiked {
+            track(withEvent: .notificationsCommentLiked) { comment in
+                CommentAnalytics.trackCommentUnLiked(comment: comment)
+            }
+        } else {
+            track(withEvent: .notificationsCommentLiked) { comment in
+                CommentAnalytics.trackCommentLiked(comment: comment)
+            }
+        }
+
+        self.state = .approved(liked: !initialIsLiked)
+        commentService.toggleLikeStatus(for: comment, siteID: siteID, success: { [weak self] in
+            guard let self, let notification = self.notification else {
+                return
+            }
+            let mediator = NotificationSyncMediator()
+            mediator?.invalidateCacheForNotification(notification.notificationId, completion: {
+                mediator?.syncNote(with: notification.notificationId)
+            })
+        }, failure: { _ in
+            self.state = .approved(liked: initialIsLiked)
+        })
+    }
+}
+
+private extension CommentModerationViewModel {
+    func approveComment() {
+        track(withEvent: .notificationsCommentApproved) { comment in
+            CommentAnalytics.trackCommentApproved(comment: comment)
+        }
+
+        isLoading = true
+        coordinator.didSelectOption()
+        commentService.approve(comment, success: { [weak self] in
+            self?.handleStatusChangeSuccess(state: .approved(liked: false))
```

Comment:
Same comment as [this one](https://github.com/wordpress-mobile/WordPress-iOS/pull/23225/files#r1608523485)